File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;
vector<string>ans;
bool solution(vector<string> v) {
    bool answer = true;
    
    sort(v.begin(),v.end());
    
    if(v.size()==1){
        return true;
    }
    
    for(int i=0;i<v.size()-1;i++){
        
        if (v[i+1].compare(0, v[i].size(), v[i]) == 0) {
            return false;  
        }
    }
    return true;
    
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phoneBook) {
        boolean answer = true;
        
        // 12 145 1245
        // 12 1245 145
        
        Arrays.sort(phoneBook);
        
        for(int i=0;i<phoneBook.length-1;i++){
            if(phoneBook[i+1].startsWith(phoneBook[i]))
                return false;
        }
        
        return true;
    }
}